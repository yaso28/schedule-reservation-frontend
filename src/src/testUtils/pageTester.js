import { mount } from '@vue/test-utils'
import MyPage from '@/components/containers/MyPage'

class PageTester {
  async mount (pageComponent, route, mountOption) {
    await mountOption.router.push(route)
    const pageWrapper = mount(MyPage, mountOption)
    return pageWrapper.findComponent(pageComponent)
  }

  assertTitleTag () {
    expect(document.title).toMatchSnapshot()
  }

  assertMetaTags () {
    expect(document.querySelectorAll('meta[data-vue-meta]')).toMatchSnapshot()
  }
}

export default new PageTester()
