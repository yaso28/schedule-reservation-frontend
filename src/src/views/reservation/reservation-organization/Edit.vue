<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyText
            :label="$t('reservation_organization.registration_number')"
            :form="form"
            field="registration_number"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ReservationOrganizationEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const record = await this.apiService.getReservationOrganization(this.id)
        this.form = this.apiService.makeForm({
          name: record.name,
          abbreviation: record.abbreviation,
          registration_number: record.registration_number
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateReservationOrganization(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.reservation-organization.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
