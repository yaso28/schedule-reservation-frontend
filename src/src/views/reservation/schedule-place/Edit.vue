<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyNumber
            :label="$t('schedule_place.price_per_hour')"
            :form="form"
            field="price_per_hour"
            :min="0"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'SchedulePlaceEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MyNumber: require('@/components/inputs/MyNumber').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const record = await this.apiService.getSchedulePlace(this.id)
        this.form = this.apiService.makeForm({
          name: record.name,
          abbreviation: record.abbreviation,
          price_per_hour: record.price_per_hour
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateSchedulePlace(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-place.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
