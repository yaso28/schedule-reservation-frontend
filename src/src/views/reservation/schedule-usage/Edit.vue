<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="loaded"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MySwitch
            :label="$t('schedule_usage.is_public')"
            :form="form"
            field="is_public"
          />
          <MySelect
            :label="$t('reservation_organization.model')"
            :form="form"
            field="reservation_organization_id"
            :options="reservationOrganizations"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ScheduleUsageEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MySwitch: require('@/components/inputs/MySwitch').default,
    MySelect: require('@/components/inputs/MySelect').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null,
      reservationOrganizations: null
    }
  },
  computed: {
    loaded () {
      return !!(this.form && this.reservationOrganizations)
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        let record
        [this.reservationOrganizations, record] = await Promise.all([
          this.reservationOrganizations ?? this.apiService.getReservationOrganizationList(),
          this.apiService.getScheduleUsage(this.id)
        ])
        this.form = this.apiService.makeForm({
          name: record.name,
          is_public: record.is_public,
          reservation_organization_id: record.reservation_organization_id
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateScheduleUsage(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-usage.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
