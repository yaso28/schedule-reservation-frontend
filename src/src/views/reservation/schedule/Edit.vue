<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="loaded"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyDate
            :label="$t('schedule.ymd')"
            :form="form"
            field="ymd"
            :show-clear="false"
          />
          <MyTime
            :label="$t('schedule.begins_at')"
            :form="form"
            field="begins_at"
          />
          <MyTime
            :label="$t('schedule.ends_at')"
            :form="form"
            field="ends_at"
          />
          <MySelect
            :label="$t('schedule_place.model')"
            :form="form"
            field="schedule_place_id"
            :options="placeList"
          />
          <MySelect
            :label="$t('schedule_usage.model')"
            :form="form"
            field="schedule_usage_id"
            :options="usageList"
          />
          <MySelect
            :label="$t('schedule_timetable.model')"
            :form="form"
            field="schedule_timetable_id"
            :options="timetableList"
          />
          <MySelect
            :label="$t('schedule_status.model')"
            :form="form"
            field="schedule_status_id"
            :options="scheduleStatusList"
          />

          <hr class="my-4">

          <p class="h5">
            {{ $t('reservation.model') }}
          </p>
          <CCard
            v-for="(reservation, index) in form.values.reservation_list"
            :key="reservation.id"
            class="my-2"
          >
            <CCardBody>
              <MyTime
                :label="$t('schedule.begins_at')"
                :form="form"
                :field="`reservation_list.${index}.begins_at`"
              />
              <MyTime
                :label="$t('schedule.ends_at')"
                :form="form"
                :field="`reservation_list.${index}.ends_at`"
              />
              <MySelect
                :label="$t('reservation_status.model')"
                :form="form"
                :field="`reservation_list.${index}.reservation_status_id`"
                :options="reservationStatusList"
              />
              <CLink
                class="btn btn-outline-primary"
                :to="{ name: 'reservation.reservation.split', params: { id: reservation.id } }"
              >
                {{ $t('action.split') }}
              </CLink>
            </CCardBody>
          </CCard>
        </CCardBody>
        <CCardFooter>
          <CButton
            class="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ReservationScheduleEdit',
  components: {
    MyDate: require('@/components/inputs/MyDate').default,
    MyTime: require('@/components/inputs/MyTime').default,
    MySelect: require('@/components/inputs/MySelect').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null,
      placeList: null,
      usageList: null,
      timetableList: null,
      scheduleStatusList: null,
      reservationStatusList: null
    }
  },
  computed: {
    loaded () {
      return !!(this.form && this.placeList && this.usageList && this.timetableList && this.scheduleStatusList && this.reservationStatusList)
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        let scheduleRecord
        let reservationRecordList
        [this.placeList, this.usageList, this.timetableList, this.scheduleStatusList, this.reservationStatusList, scheduleRecord, reservationRecordList] = await Promise.all([
          this.placeList ?? this.apiService.getSchedulePlaceList(),
          this.usageList ?? this.apiService.getScheduleUsageList(),
          this.timetableList ?? this.apiService.getScheduleTimetableList(),
          this.scheduleStatusList ?? this.apiService.getScheduleStatusList(),
          this.reservationStatusList ?? this.apiService.getReservationStatusList(),
          this.apiService.getSchedule(this.id),
          this.apiService.getReservationListForSchedule(this.id)
        ])
        this.form = this.apiService.makeForm({
          ymd: scheduleRecord.ymd,
          begins_at: scheduleRecord.begins_at,
          ends_at: scheduleRecord.ends_at,
          schedule_place_id: scheduleRecord.schedule_place_id,
          schedule_usage_id: scheduleRecord.schedule_usage_id,
          schedule_timetable_id: scheduleRecord.schedule_timetable_id,
          schedule_status_id: scheduleRecord.schedule_status_id,
          reservation_list: reservationRecordList.map(record => ({
            id: record.id,
            begins_at: record.begins_at,
            ends_at: record.ends_at,
            reservation_status_id: record.reservation_status_id
          }))
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateSchedule(this.id, this.form)
        this.$store.dispatch('alert/saveSuccessNoRedirect')
        this.form.setInit()
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
