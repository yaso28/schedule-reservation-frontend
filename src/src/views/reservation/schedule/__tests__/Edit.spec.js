import Edit from '../Edit'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import promiseMocker from '@/testUtils/promiseMocker'
import storeModules from '@/store/modules'
import myFormMocker from '@/testUtils/myFormMocker'

jest.mock('@/store/modules')
jest.mock('@/models/MyForm')

const placeList = require('@/testUtils/dummyApiData/schedule-place/list.json')
const usageList = require('@/testUtils/dummyApiData/schedule-usage/list.json')
const timetableList = require('@/testUtils/dummyApiData/schedule-timetable/list.json')
const scheduleStatusList = require('@/testUtils/dummyApiData/schedule-status/list.json')
const reservationStatusList = require('@/testUtils/dummyApiData/reservation-status/list.json')
const scheduleRecord = require('@/testUtils/dummyApiData/schedule/get.json')
const reservationRecordList = require('@/testUtils/dummyApiData/reservation/list-for-schedule-split.json')
const id = scheduleRecord.id
const makeFormArgs = {
  ymd: scheduleRecord.ymd,
  begins_at: scheduleRecord.begins_at,
  ends_at: scheduleRecord.ends_at,
  schedule_place_id: scheduleRecord.schedule_place_id,
  schedule_usage_id: scheduleRecord.schedule_usage_id,
  schedule_timetable_id: scheduleRecord.schedule_timetable_id,
  schedule_status_id: scheduleRecord.schedule_status_id,
  reservation_list: reservationRecordList.map(record => ({
    id: record.id,
    begins_at: record.begins_at,
    ends_at: record.ends_at,
    reservation_status_id: record.reservation_status_id
  }))
}
const saveResult = require('@/testUtils/dummyApiData/common/update.json')
saveResult.id = id

describe('Edit', () => {
  let mountOption
  let apiService
  beforeEach(() => {
    promiseMocker.init()
    mountOption = mountOptionCreator.create(storeModules)
    apiService = mountOption.provide.apiService
  })

  afterEach(() => {
    promiseMocker.clear()
    jest.resetAllMocks()
  })

  describe('マウント', () => {
    let wrapper
    let loadError
    let loadPlaceListSwitch
    let loadUsageListSwitch
    let loadTimetableListSwitch
    let loadScheduleStatusListSwitch
    let loadReservationStatusListSwitch
    let loadScheduleRecordSwitch
    let loadReservationRecordListSwitch
    beforeEach(async () => {
      loadError = new Error('load')
      loadPlaceListSwitch = promiseMocker.createSwitch()
      loadUsageListSwitch = promiseMocker.createSwitch()
      loadTimetableListSwitch = promiseMocker.createSwitch()
      loadScheduleStatusListSwitch = promiseMocker.createSwitch()
      loadReservationStatusListSwitch = promiseMocker.createSwitch()
      loadScheduleRecordSwitch = promiseMocker.createSwitch()
      loadReservationRecordListSwitch = promiseMocker.createSwitch()
      apiService.getSchedulePlaceList = promiseMocker.mock(placeList, loadError, loadPlaceListSwitch)
      apiService.getScheduleUsageList = promiseMocker.mock(usageList, loadError, loadUsageListSwitch)
      apiService.getScheduleTimetableList = promiseMocker.mock(timetableList, loadError, loadTimetableListSwitch)
      apiService.getScheduleStatusList = promiseMocker.mock(scheduleStatusList, loadError, loadScheduleStatusListSwitch)
      apiService.getReservationStatusList = promiseMocker.mock(reservationStatusList, loadError, loadReservationStatusListSwitch)
      apiService.getSchedule = promiseMocker.mock(scheduleRecord, loadError, loadScheduleRecordSwitch)
      apiService.getReservationListForSchedule = promiseMocker.mock(reservationRecordList, loadError, loadReservationRecordListSwitch)

      wrapper = await pageTester.mount(Edit, { name: 'reservation.schedule.edit', params: { id } }, mountOption)
    })

    test('titleタグ', () => {
      pageTester.assertTitleTag()
    })

    test('metaタグ', () => {
      pageTester.assertMetaTags()
    })

    test('Processingカバー表示開始', () => {
      expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
    })

    test.each([
      ['場所', apiService => apiService.getSchedulePlaceList],
      ['用途', apiService => apiService.getScheduleUsageList],
      ['時間割', apiService => apiService.getScheduleTimetableList],
      ['公開ステータス', apiService => apiService.getScheduleStatusList],
      ['予約ステータス', apiService => apiService.getReservationStatusList]
    ])('%sマスタ取得処理を呼出', (masterName, getApiMethodToAssert) => {
      expect(getApiMethodToAssert(apiService)).toHaveBeenCalledWith()
    })

    test('予定レコード取得処理を呼出', () => {
      expect(apiService.getSchedule).toHaveBeenCalledWith(id)
    })

    test('予約レコード取得処理を呼出', () => {
      expect(apiService.getReservationListForSchedule).toHaveBeenCalledWith(id)
    })

    test('Processingカバー表示中', () => {
      expect(
        storeModules.processing.mutations.endGet
      ).not.toHaveBeenCalled()
    })

    describe.each([
      ['場所マスタ'],
      ['用途マスタ'],
      ['時間割マスタ'],
      ['公開ステータスマスタ'],
      ['予約ステータスマスタ'],
      ['予定レコード'],
      ['予約レコード']
    ])('%s取得：エラー', (dataName) => {
      beforeEach(() => {
        const loadSwitch =
          dataName === '場所マスタ' ? loadPlaceListSwitch
            : dataName === '用途マスタ' ? loadUsageListSwitch
              : dataName === '時間割マスタ' ? loadTimetableListSwitch
                : dataName === '公開ステータスマスタ' ? loadScheduleStatusListSwitch
                  : dataName === '予約ステータスマスタ' ? loadReservationStatusListSwitch
                    : dataName === '予定レコード' ? loadScheduleRecordSwitch
                      : dataName === '予約レコード' ? loadReservationRecordListSwitch
                        : undefined
        promiseMocker.setSwitch(loadSwitch, false)
        apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
        promiseMocker.flushPendings()
      })

      test('エラー処理', () => {
        expect(apiService.isHandledError).toHaveBeenCalledWith(loadError)
      })

      test('Processingカバー表示終了', () => {
        expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
      })
    })

    describe('データ取得：成功', () => {
      let form
      beforeEach(() => {
        form = myFormMocker.createMock(makeFormArgs)
        apiService.makeForm = jest.fn().mockReturnValueOnce(form)
        promiseMocker.flushPendings()
      })

      test('フォーム作成', () => {
        expect(apiService.makeForm).toHaveBeenCalledWith(makeFormArgs)
      })

      test('ページ全体表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })

      test('Processingカバー表示終了', () => {
        expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
      })

      describe('URL変更', () => {
        let newId
        beforeEach(() => {
          newId = id + 1
          apiService.getSchedulePlaceList = promiseMocker.mockResolved(placeList)
          apiService.getScheduleUsageList = promiseMocker.mockResolved(usageList)
          apiService.getScheduleTimetableList = promiseMocker.mockResolved(timetableList)
          apiService.getScheduleStatusList = promiseMocker.mockResolved(scheduleStatusList)
          apiService.getReservationStatusList = promiseMocker.mockResolved(reservationStatusList)
          apiService.getSchedule = promiseMocker.mockResolved(scheduleRecord)
          apiService.getReservationListForSchedule = promiseMocker.mockResolved(reservationRecordList)

          storeModules.processing.mutations.beginGet.mockReset()
          storeModules.processing.mutations.endGet.mockReset()
          wrapper.setProps({ id: newId })
        })

        test('Processingカバー表示開始', () => {
          expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
        })

        test.each([
          ['場所', apiService => apiService.getSchedulePlaceList],
          ['用途', apiService => apiService.getScheduleUsageList],
          ['時間割', apiService => apiService.getScheduleTimetableList],
          ['公開ステータス', apiService => apiService.getScheduleStatusList],
          ['予約ステータス', apiService => apiService.getReservationStatusList]
        ])('%sマスタ取得処理は呼び出さない', (masterName, getApiMethodToAssert) => {
          expect(getApiMethodToAssert(apiService)).not.toHaveBeenCalled()
        })

        test('予定レコード取得処理を呼出', () => {
          expect(apiService.getSchedule).toHaveBeenCalledWith(newId)
        })

        test('予約レコード取得処理を呼出', () => {
          expect(apiService.getReservationListForSchedule).toHaveBeenCalledWith(newId)
        })

        test('Processingカバー表示中', () => {
          expect(
            storeModules.processing.mutations.endGet
          ).not.toHaveBeenCalled()
        })
      })

      describe('リセットボタンクリック', () => {
        beforeEach(() => {
          wrapper.find('button.btn-reset').trigger('click')
        })

        test('フォームリセット', () => {
          expect(form.reset).toHaveBeenCalledWith()
        })
      })

      describe('フォーム送信', () => {
        let saveError
        let saveSwitch
        beforeEach(() => {
          saveError = new Error('save')
          saveSwitch = promiseMocker.createSwitch()
          apiService.updateSchedule = promiseMocker.mock(saveResult, saveError, saveSwitch)
          wrapper.find('form').trigger('submit')
        })

        test('Processingカバー表示開始', () => {
          expect(storeModules.processing.mutations.beginPost).toHaveBeenCalled()
        })

        test('アラートクリア', () => {
          expect(storeModules.alert.mutations.clear).toHaveBeenCalled()
        })

        test('保存処理を呼出', () => {
          expect(apiService.updateSchedule).toHaveBeenCalledWith(id, form)
        })

        test('Processingカバー表示中', () => {
          expect(storeModules.processing.mutations.endPost).not.toHaveBeenCalled()
        })

        describe('保存：エラー', () => {
          beforeEach(() => {
            promiseMocker.setSwitch(saveSwitch, false)
            apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
            promiseMocker.flushPendings()
          })

          test('エラー処理', () => {
            expect(apiService.isHandledError).toHaveBeenCalledWith(saveError)
          })

          test('Processingカバー表示終了', () => {
            expect(storeModules.processing.mutations.endPost).toHaveBeenCalled()
          })

          test('成功アラートは呼び出さない', () => {
            expect(storeModules.alert.actions.saveSuccessNoRedirect).not.toHaveBeenCalled()
          })

          test('フォーム初期値更新は呼び出さない', () => {
            expect(form.setInit).not.toHaveBeenCalled()
          })
        })

        describe('保存：成功', () => {
          beforeEach(() => {
            promiseMocker.setSwitch(saveSwitch, true)
            promiseMocker.flushPendings()
          })

          test('成功アラート', () => {
            expect(storeModules.alert.actions.saveSuccessNoRedirect).toHaveBeenCalled()
          })

          test('フォーム初期値更新', () => {
            expect(form.setInit).toHaveBeenCalledWith()
          })

          test('Processingカバー表示終了', () => {
            expect(storeModules.processing.mutations.endPost).toHaveBeenCalled()
          })
        })
      })
    })
  })
})
