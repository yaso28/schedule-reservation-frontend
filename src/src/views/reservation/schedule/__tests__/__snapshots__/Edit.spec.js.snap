// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit マウント metaタグ 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="2"
    name="robots"
  />,
]
`;

exports[`Edit マウント titleタグ 1`] = `"やそスポーツクラブ | 練習予定編集"`;

exports[`Edit マウント データ取得：成功 ページ全体表示 1`] = `
<div>
  <h1>
    練習予定編集
  </h1>
   
  <form
    class=""
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="form-row form-group"
        >
          <label
            class="col-form-label col-sm-3"
            for="ymd"
          >
            
    日付
  
          </label>
           
          <div
            class="col-sm-9"
          >
            <div
              class="my-input-wrapper"
            >
              <div
                class="d-flex"
              >
                <input
                  class="form-control flex-grow-1"
                  id="ymd"
                  type="date"
                />
                 
                <!---->
              </div>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="begins_at"
          >
             開始時刻 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <input
              class="form-control"
              id="begins_at"
              step="900"
              type="time"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="ends_at"
          >
             終了時刻 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <input
              class="form-control"
              id="ends_at"
              step="900"
              type="time"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="schedule_place_id"
          >
             場所 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <select
              class="form-control"
              id="schedule_place_id"
            >
              <!---->
              <option
                data-key="0"
                value=""
              >
                  
              </option>
              <option
                data-key="1"
                value="1"
              >
                 体育館＠公民館別棟 
              </option>
              <option
                data-key="2"
                value="2"
              >
                 多目的ホール＠公民館1F 
              </option>
              <option
                data-key="3"
                value="3"
              >
                 大会議室＠公民館2F 
              </option>
            </select>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="schedule_usage_id"
          >
             用途 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <select
              class="form-control"
              id="schedule_usage_id"
            >
              <!---->
              <option
                data-key="0"
                value=""
              >
                  
              </option>
              <option
                data-key="1"
                value="1"
              >
                 通常練習 
              </option>
              <option
                data-key="2"
                value="2"
              >
                 体験会 
              </option>
              <option
                data-key="3"
                value="3"
              >
                 会議 
              </option>
            </select>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="schedule_timetable_id"
          >
             時間割 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <select
              class="form-control"
              id="schedule_timetable_id"
            >
              <!---->
              <option
                data-key="0"
                value=""
              >
                  
              </option>
              <option
                data-key="1"
                value="1"
              >
                 午前 
              </option>
              <option
                data-key="2"
                value="2"
              >
                 午後 
              </option>
              <option
                data-key="3"
                value="3"
              >
                 一日 
              </option>
            </select>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="schedule_status_id"
          >
             公開ステータス 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <select
              class="form-control"
              id="schedule_status_id"
            >
              <!---->
              <option
                data-key="0"
                value=""
              >
                  
              </option>
              <option
                data-key="1"
                value="1"
              >
                 未定 
              </option>
              <option
                data-key="2"
                value="2"
              >
                 確定 
              </option>
              <option
                data-key="3"
                value="3"
              >
                 中止 
              </option>
              <option
                data-key="4"
                value="4"
              >
                 非公開 
              </option>
            </select>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <hr
          class="my-4"
        />
         
        <p
          class="h5"
        >
          
          会場予約
        
        </p>
         
        <div
          class="card my-2"
        >
          <div
            class="card-body"
          >
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.0.begins_at"
              >
                 開始時刻 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <input
                  class="form-control"
                  id="reservation_list.0.begins_at"
                  step="900"
                  type="time"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.0.ends_at"
              >
                 終了時刻 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <input
                  class="form-control"
                  id="reservation_list.0.ends_at"
                  step="900"
                  type="time"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.0.reservation_status_id"
              >
                 予約ステータス 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <select
                  class="form-control"
                  id="reservation_list.0.reservation_status_id"
                >
                  <!---->
                  <option
                    data-key="0"
                    value=""
                  >
                      
                  </option>
                  <option
                    data-key="1"
                    value="1"
                  >
                     未予約 
                  </option>
                  <option
                    data-key="2"
                    value="2"
                  >
                     電話予約済 
                  </option>
                  <option
                    data-key="3"
                    value="3"
                  >
                     書類提出済 
                  </option>
                  <option
                    data-key="4"
                    value="4"
                  >
                     書類控取得 
                  </option>
                  <option
                    data-key="5"
                    value="5"
                  >
                     料金支払済 
                  </option>
                  <option
                    data-key="6"
                    value="6"
                  >
                     会計精算済 
                  </option>
                  <option
                    data-key="7"
                    value="7"
                  >
                     キャンセル 
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <a
              class="btn btn-outline-primary"
              href="/reservation/reservation/split/8"
              target="_self"
            >
              
              分割
            
            </a>
          </div>
        </div>
        <div
          class="card my-2"
        >
          <div
            class="card-body"
          >
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.1.begins_at"
              >
                 開始時刻 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <input
                  class="form-control"
                  id="reservation_list.1.begins_at"
                  step="900"
                  type="time"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.1.ends_at"
              >
                 終了時刻 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <input
                  class="form-control"
                  id="reservation_list.1.ends_at"
                  step="900"
                  type="time"
                />
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <div
              class="form-group form-row"
              role="group"
            >
              <label
                class="col-form-label col-sm-3"
                for="reservation_list.1.reservation_status_id"
              >
                 予約ステータス 
              </label>
              <div
                class="col-sm-9"
              >
                <!---->
                <select
                  class="form-control"
                  id="reservation_list.1.reservation_status_id"
                >
                  <!---->
                  <option
                    data-key="0"
                    value=""
                  >
                      
                  </option>
                  <option
                    data-key="1"
                    value="1"
                  >
                     未予約 
                  </option>
                  <option
                    data-key="2"
                    value="2"
                  >
                     電話予約済 
                  </option>
                  <option
                    data-key="3"
                    value="3"
                  >
                     書類提出済 
                  </option>
                  <option
                    data-key="4"
                    value="4"
                  >
                     書類控取得 
                  </option>
                  <option
                    data-key="5"
                    value="5"
                  >
                     料金支払済 
                  </option>
                  <option
                    data-key="6"
                    value="6"
                  >
                     会計精算済 
                  </option>
                  <option
                    data-key="7"
                    value="7"
                  >
                     キャンセル 
                  </option>
                </select>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
             
            <a
              class="btn btn-outline-primary"
              href="/reservation/reservation/split/7"
              target="_self"
            >
              
              分割
            
            </a>
          </div>
        </div>
      </div>
       
      <footer
        class="card-footer"
      >
        <button
          class="btn btn-reset btn-secondary"
          type="button"
        >
          
          リセット
        
        </button>
         
        <button
          class="btn btn-primary"
          type="submit"
        >
          
          保存
        
        </button>
      </footer>
    </div>
  </form>
</div>
`;
